apps/OboeTester/app/src/main/cpp/NativeAudioContext.cpp

```diff
@@ -518,6 +547,21 @@ void ActivityTestOutput::runBlockingIO() {
             LOGE("%s() : write() wrote %d of %d\n", __func__, framesWritten, framesPerBlock);
             break;
         }
+
+        const int64_t bufferSizeInFrames = mBufferSizeInFrames.load();
+        if (oboeStream->getPerformanceMode() == PerformanceMode::POWER_SAVING_OFFLOADED &&
+            bufferSizeInFrames > mSampleRate) {
+            // If it is offload stream, the buffer size is more than 1 second and it is almost full,
```

Comment:
where does the 1 second come from? It's not obvious from reading this code

Reply:
As I mentioned offline, this is a naive assumption that the OS won't suspend if there is any CPU work within one second. If the OS is not suspended, the audio framework won't suspend to drain data. There won't be issues to try to write more data. One other reason is that we use one second as the writing time out so that if we care about writing time out, at least one second of buffer size will be a reasonable minimum requirement.

Reply:
Ohhh, bufferSizeInFrames > mSampleRate is one second...